package.json

```diff
@@ -55,7 +55,9 @@
     "npm": ">=7.0.0"
   },
   "scripts": {
-    "test": "run-s format lint type-check build unit-test",
+    "test": "run-s format lint type-check build test:*",
+    "test:unit": "jest",
+    "test:types": "tstyche",
```

Comment:
I added `test:types` script to make it easy to pass extra options. For example, in CI the command could be `test:types --target 4.5,5.0,current`. Just a suggestion, it can be set up somehow else.

Currently `type-check` script is used. In early stages I saw TSTyche as a **ty**pe **che**cker (this is where the name is coming from), but it does more. It uses TypeScript to perform static analysis and file are not executed, but with `.only` / `.skip` and other features it feels more like a **type test runner**.

Reply:
Actually, we do have [this file](https://github.com/immutable-js/immutable-js/blob/d7664bf9d3539da8ea095f2ed08bbe1cd0d46071/type-definitions/ts-tests/index.d.ts#L1) telling that the minimum version is 4.5, and I think that we do only test 4.5 (not really sure about that, we might use the local typescript version).

It might be interesting to test multiple typescript version in CI (but not done actually, so if it can be do, it's great, but not a necessity).

Locally, I think that testing the local version of typescript is good enough.

Reply:
Right. I forgot to mention that `current` is currently (or locally) installed TypeScript. This is the default. I mean, `tstyche` is the same as `tstyche --target current`.